### PR TITLE
GitHub Actions workflow for building and pushing multi-platform Docker images to Docker Hub

### PR DESCRIPTION
**Pull request by joel-ling**
Merging this pull request would add two Actions workflows:

- "Build and push semver-tagged Docker image"
- "Build and push Docker image tagged `main`"

The first of the two workflows runs automatically whenever a tag of the following format is pushed: `v[0-9]+.[0-9]+.[0-9]+`. The second is run whenever the `main` branch is pushed to.

Both workflows build and push images for the platforms specified by the Actions variable `PLATFORMS`.

Image tags are formatted as follows:

- `<ORGANISATION_NAME>/<IMAGE_NAME>:<MAJOR>.<MINOR>.<PATCH>` (and `latest`), or
- `<ORGANISATION_NAME>/<IMAGE_NAME>:main`, as may be the case.

The Docker Hub organisation name, username, and access token are automatically populated with GitHub Actions Secrets, while the image name is an Actions variable of the repository.

### CODE REVIEW

File: .github/workflows/build-image-master.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v6
+        with:
+          platforms: ${{ vars.PLATFORMS }}
+          push: true
+          sbom: true
+          tags: >-
+            ${{ secrets.DOCKERHUB_ORGANIZATION_NAME }}/${{ vars.IMAGE_NAME }}:master
```
Comment (neilcook) [inline .github/workflows/build-image-master.yml:35]: Given that we have a `main` branch not a `master` branch, using the `master` tag feels a little weird.

**Review by neilcook — APPROVED** [review]
LGTM, just a comment about the master tag.

I guess this is hard to test without actually pushing to the main branch or tagging a release.

File: .github/workflows/build-image-master.yml (hunk 1 above)
Comment (joel-ling) [reply .github/workflows/build-image-master.yml]: Thanks, @neilcook!

> Given that we have a `main` branch not a `master` branch, using the `master` tag feels a little weird.

I have changed the tag from `master` to `main`.

> I guess this is hard to test without actually pushing to the main branch or tagging a release.

Not to worry, the workflows are passing in the fork (we are about to merge), albeit using alternative variables and secrets, hence pushing to a different repository on Docker Hub.

**Comment by joel-ling** [comment]
Confirmed that `powerdns/lightningstream:main` is available on [Docker Hub](https://hub.docker.com/r/powerdns/lightningstream/tags). `powerdns/lightningstream:v0.4.4` or `powerdns/lightningstream:v0.5.0` (and `powerdns/lightningstream:latest`) should automatically become available upon the next release.

**Pull request merged**